packages/block-editor/src/components/block-controls/slot.js

```diff
@@ -31,8 +34,55 @@ export default function BlockControlsSlot( { group = 'default', ...props } ) {
 		[ toolbarState, contextState ]
 	);
 
+	// Create filter function for content-only mode
+	const filter = useMemo( () => {
+		if ( blockEditingMode !== 'contentOnly' ) {
+			return undefined; // No filtering in default mode
+		}
+
+		return ( fill ) => {
+			// Check if the fill has category="content"
+			if ( fill?.children && typeof fill.children === 'function' ) {
+				// For function children, we need to check the rendered result
+				const renderedChildren = fill.children( fillProps );
+				return hasCategoryContent( renderedChildren );
+			} else if ( fill?.children ) {
+				// For React element children, check directly
+				return hasCategoryContent( fill.children );
+			}
+			return false;
+		};
+	}, [ blockEditingMode, fillProps ] );
+
+	// Helper function to check if any component has category="content"
+	const hasCategoryContent = ( children ) => {
+		if ( ! children ) {
+			return false;
+		}
+
+		if ( Array.isArray( children ) ) {
+			return children.some( ( child ) => {
+				// Check if this component has category="content"
+				if ( child?.props?.category === 'content' ) {
+					return true;
+				}
+
+				// Recursively check children
+				return hasCategoryContent( child?.props?.children );
+			} );
+		}
+
+		// Check if this component has category="content"
+		if ( children?.props?.category === 'content' ) {
+			return true;
+		}
+
+		// Recursively check children
+		return hasCategoryContent( children?.props?.children );
+	};
```

Comment:
I don't think we could launch a recursive prop check. There's got to be a better way. This does work as a POC though.

Reply:
Actually... I don't think this is filtering right. Surely there is a filtering function that could work?